### PR TITLE
UI: Adds dashboard page layout

### PR DESCRIPTION
**Pull request by pratap0007**
Signed-off-by: Shiv Verma <shverma@redhat.com>

# Changes

 This patch includes following component to create Homepage layout
    - Header
    - Footer
    - SearchBar
    - Left Pane to Filter Resources
    - Resources for rendering resources
    - Details for rendering resource detail
    - BackGround for Homepage banner
    
 Signed-off-by: Shiv Verma <shverma@redhat.com>

Here, have added screenshot of Homepage layout

image

image

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/App/index.tsx [hunk 1]
```diff
@@ -3,20 +3,40 @@ import { observer } from 'mobx-react';
...
-            <LeftPane />
-          </GridItem>
-        </Grid>
-      </div>
+      <Router>
+        <Page header={<Header />}>
+          <Route exact path="/" component={Background} />
+          <PageSection>
+            <Grid hasGutter>
+              <Grid>
```
Comment (pratap0007) [inline ui/src/containers/App/index.tsx:25]: should it be GridItem instead of Grid ?

File: ui/src/containers/Details/index.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { GridItem } from '@patternfly/react-core';
+
+const Details: React.FC = () => {
+  return (
+    <React.Fragment>
+      <GridItem span={12}>Add Resources detail here</GridItem>
```
Comment (pratap0007) [inline ui/src/containers/Details/index.tsx:7]: no need of GridItem here

File: ui/src/containers/App/App.css [hunk 3]
```diff
@@ -0,0 +1,4 @@
+.pf-c-page {
+  position: fixed;
+  width: 100%;
```
Comment (pratap0007) [inline ui/src/containers/App/App.css:3]: I guess without width  works same

File: ui/src/containers/Background/Background.css [hunk 4]
```diff
@@ -0,0 +1,17 @@
+.hub-background-image {
+  background-image: url('../../assets/logo/background.jpg');
+  background-repeat: no-repeat;
+  background-attachment: fixed;
+  background-size: cover;
+  padding: 2em;
+}
+
+.hub-heading {
```
Comment (pratap0007) [inline ui/src/containers/Background/Background.css:9]: .hub-background-heading , wdyt?

File: ui/src/containers/Footer/index.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+import '@patternfly/react-core/dist/styles/base.css';
+import tekton from '../../assets/logo/tekton.png';
+import './Footer.css';
+const Footer: React.FC = () => {
+  return (
+    <React.Fragment>
+      <Card className="hub-footer-card">
+        <Grid>
+          <GridItem span={12} className="text">
+            <a href="https://cd.foundation">
```
Comment (pratap0007) [inline ui/src/containers/Footer/index.tsx:12]: all links should be open in a new tab,right?

File: ui/src/containers/Resources/index.tsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+  ) => {
+    setPerPage(perpage);
+  };
+  return (
+    <React.Fragment>
+      <Grid>
+        <GridItem>
+          <Pagination
+            itemCount={200}
+            perPage={perPgae}
```
Comment (vinamra28) [inline ui/src/containers/Resources/index.tsx:34]: are we going to include pagination in this release ? (if not then probably we can remove this for now)

**Review by vinamra28 — CHANGES_REQUESTED** [review]

File: ui/src/containers/Resources/index.tsx (hunk 6 above)
Comment (PuneetPunamiya) [reply ui/src/containers/Resources/index.tsx:34]: Yes in the follow up pr it is been removed

File: ui/src/containers/App/App.test.tsx [hunk 7]
```diff
@@ -14,18 +14,22 @@ describe('App', () => {
...
       </Provider>
     );
 
     expect(app.toJSON()).toMatchSnapshot();
     done();
   });
 
-  it('should find the leftpane component and match the count', () => {
-    const component = shallow(<App />);
+  it('should find the component and match the count', () => {
```
Comment (vinamra28) [inline ui/src/containers/App/App.test.tsx:25]:
```suggestion
  it('should find the left pane component and match the count', () => {
```

File: ui/src/containers/Resources/index.tsx [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+    setPageNumber(page);
+  };
+
+  const perPageSelect = (
+    event: React.MouseEvent | React.KeyboardEvent | MouseEvent,
+    perpage: number
+  ) => {
+    setPerPage(perpage);
+  };
+  return (
```
Comment (vinamra28) [inline ui/src/containers/Resources/index.tsx:28]: can we add a space in between?

**Pull request merged**